src/core/agent/modules/signify/signifyApi.ts

```diff
@@ -81,14 +100,26 @@ export class SignifyApi {
     this.opRetryInterval = opRetryInterval;
   }
 
-  async bootAndConnect(retryInterval = 1000) {
+  async bootAndConnect(
+    retryInterval = 1000,
+    maximumNumberOfRetries?: number,
+    numberOfRetries = 0
+  ) {
     try {
+      SignifyApi.isOnline = false;
       await this.signifyClient.boot();
       await this.signifyClient.connect();
       SignifyApi.isOnline = true;
     } catch (error) {
+      if (maximumNumberOfRetries && numberOfRetries >= maximumNumberOfRetries) {
+        throw new Error(SignifyApi.FAILED_TO_ESTABLISH_KERIA_CONNECTION);
+      }
```

Comment:
We don't need this retry limit - we want to retry forever while the app is open. We can implement backoff/jitter in another story for the timing.